utils/windows-utils.ts

```diff
@@ -11,4 +14,106 @@ export function validatePackageID(id: string) {
+  parts.push("edge");
+  return parts.join(".");
+}
+
+export function validateWindowsOptions(options) {
+  const validationErrors: { field: keyof WindowsPackageOptions | null, error: string }[] = [];
+  if (!options) {
+    validationErrors.push({ field: null, error: "No options specified " });
+    return validationErrors;
+  }
+
+  if (!options.packageId) {
+    validationErrors.push({ field: "packageId", error: "No package ID" });
+  }
+
+  if (!options.name || options.name.trim().length === 0) {
+    validationErrors.push({ field: "name", error: "Must have an app name" });
+  }
+  else if (options.name.trim().length > 256) {
+    validationErrors.push({ field: "name", error: "App name must be less than 256 characters" });
+  }
+
+  if (!options.version || options.version.trim().length === 0 && !options.classicPackage.version || options.classicPackage.version.trim().length === 0) {
+    validationErrors.push({ field: "version", error: "Must have an app version and a Classic Package version" });
+  }
+  else {
+    validateWindowsPackageVersion(options.version, options.classicPackage.version, validationErrors);
+  }
+
+  // Validating publisher options when we have publisher data
+  if (!options.publisher) {
+    validationErrors.push({ field: "publisher", error: "Publisher information must be supplied." });
+  } else {
+    // commonName and displayName are required, all other publisher fields are optional
+    const requiredPublisherFields: Array<keyof WindowsPublisherOptions> = [
+      "displayName",
+      "commonName"
+    ];
+
+    requiredPublisherFields
+      .filter(prop => !options.publisher![prop])
+      .forEach(() => validationErrors.push({ field: "publisher", error: `Publisher info must be specified` }));
+
+    // Ensure country code is 2 chars
+    if (options.publisher.country && options.publisher.country.length !== 2) {
+      validationErrors.push({ field: "publisher", error: "Publisher country code must be 2 letters" });
+    }
+  }
+
+  if (!options.url) {
+    validationErrors.push({ field: "url", error: "Start URL must be specified. If your start URL is the same as Host, you can use '/' as the start URL." });
+  } else {
+    const startUrlError = validateUrl(options.url, options.host);
+    if (startUrlError) {
+      validationErrors.push({ field: "url", error: "URL is invalid" });
+    }
+  }
+
+  return validationErrors;
+}
+
+function validateWindowsPackageVersion(version: string, classicVersion: string, validationErrors: { field: keyof WindowsPackageOptions | null, error: string }[]) {
+  console.log(version.trim(), version.trim().length);
+
+  /**
+   * Version must be in this form: "1.0.0"; a 3 segment version. 
+   * 4 segment versions (e.g. 1.0.0.0) are invalid due to Store restrictions; 
+   * the Store reserves the 4th segment for internal use.
+   * 
+   * versions in the 3 segment version should always only have 5 characters
+   */
+  if (version.trim().length !== 5) {
```

Comment:
I don't think this will work; this is a valid version with more than 5 characters: 1.32.712

Reply:
Ahh ok, yeah need to do more thinking on this

Reply:
Ok, changed this check to split the version by decimal point and then check the length of that array. This works with numbers such as 1.32.712 .